wormhole-connect/src/config/types.ts

```diff
@@ -363,3 +367,54 @@ export type NttGroup = {
 };
 
 export type NttGroups = { [key: string]: NttGroup };
+
+// Token bridge foreign asset cache
+// Used in utils/sdkv2.ts
+
+type ForeignAssets<C extends ChainV2> = Record<string, TokenAddressV2<C>>;
+
+export class WrappedTokenAddressCache {
+  caches: Partial<Record<ChainV2, ForeignAssets<ChainV2>>>;
+
+  constructor(tokens: TokensConfig, converter: SDKConverter) {
+    this.caches = {};
+
+    // Pre-populate cache with values from built-in config
+    for (const key in tokens) {
+      const token = tokens[key];
+
+      for (const chain in token.foreignAssets) {
+        const foreignAsset = tokens[key].foreignAssets![chain];
+        const chainV2 = converter.toChainV2(chain as ChainName);
+        const addr = WormholeV2.parseAddress(chainV2, foreignAsset.address);
+        this.set(key, chainV2, addr);
+      }
+
+      // Cache it on its native chain too
+      if (token.tokenId) {
+        const nativeChainV2 = converter.toChainV2(token.nativeChain);
+        this.set(
+          key,
+          nativeChainV2,
+          WormholeV2.parseAddress(nativeChainV2, token.tokenId!.address),
```

Comment:
Nitpick: Do you need `token.tokenId!` as you already checked `token.tokenId` in the if condition?

Reply:
youre right